rustls/src/client/mod.rs

```diff
@@ -201,7 +201,7 @@ pub enum ServerName {
     /// The server is identified by a DNS name.  The name
     /// is sent in the TLS Server Name Indication (SNI)
     /// extension.
-    DnsName(webpki::DnsName),
+    DnsName(DnsName),
```

Comment:
A `non_exhaustive` enum with public variants is already a bad and inconvenient API for callers. I think we should have `pub struct ServerName(ServerNameInner)` instead, where this enum is renamed to `ServerNameInner`.

Consider in particular that it is likely that `webpki` will have its own `IpAddress` type unless/until `std::net::IpAddr` can be moved to core.

Reply:
OK, having looked at the next commit, I'll backtrack.

> A non_exhaustive enum with public variants is already a bad and inconvenient API for callers. I think we should have pub struct ServerName(ServerNameInner) instead, where this enum is renamed to ServerNameInner.

I think we probably should do that at some point in the future, for the reason given, but elsewhere we already need to have a `DnsName` type in Rustls that we can expose in the public API.